packages/uniforms/__suites__/TextField.tsx

```diff
@@ -0,0 +1,126 @@
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import createContext, { render } from './renderWithContext';
+
+export function TextFieldTests(TextField: React.ComponentType<any>) {
```

Comment:
The typing could be improved but I was having trouble finding an elegant way to get the typings that `connectField` returns. Any suggestions/ideas/hints are welcome.

Reply:
On one hand, you can go with something like `Component` from the `createAutoField` helper. On the other, something like `ComponentType<GuaranteedProps<Value>>` (for `TextField`, `Value=string`).

Reply:
`ComponentType<GuaranteedProps<Value>>` reports errors with missing properties that are not passed because they are needed in some tests and `Partial` unfortunately doesn't work for `autoComplete` and `type` props. Any other ideas?

Reply:
Then let's leave it as it is for now.